deploy/environments/applications_nse.yaml

```diff
@@ -0,0 +1,72 @@
+spack:
+  view: false
+  concretization: separately
+  modules:
+    tcl:
+      all:
+        environment:
+          set:
+            '${PACKAGE}_ROOT': '${PREFIX}'
+      whitelist:
+        - circuit-analysis
+        - connectome-tools
+        - emsim
+        - psp-validation
+        - py-atlas-building-tools
+        - py-bbp-analysis-framework
+        - py-bbp-workflow
+        - py-bluepy
+        - py-bluepysnap
+        - py-currentscape
+        - py-entity-management
+        - py-minis-validation
+        - py-morph-tool
+        - py-morphio
+        - py-morphology-repair-workflow
+        - py-simwriter
+        - py-sonata-network-reduction
+        - regiodesics
+      projections:
+        all: '{name}/{version}'
+  packages:
+    all:
+      compiler: [gcc@9.3.0, intel]
+      providers:
+        mpi: [hpe-mpi@2.21]
+        lapack: [intel-mkl@2019.3.199]
+    hdf5:
+      variants: ~mpi+hl
+    highfive:
+      variants: ~mpi
+    libsonata:
+      variants: ~mpi
+    mvdtool:
+      variants: ~mpi
+    netcdf-c:
+      variants: ~mpi
+    py-h5py:
+      variants: ~mpi
+    py-mvdtool:
+      variants: ~mpi
+    python:
+      version: [3.8.3]
+    vtk:
+      variants: ~mpi~ffmpeg~opengl2
```

Comment:
@matz-e : Is this still WIP / experimental or ready for review?

I am curious about some of the aspects : e.g. different application stacks can "prefer" different version/variants of package. This is good if this approach is going to avoid the conflict from +mpi vs ~mpi variant aspects.

But I assume this doesn't change much when we have to combine nse + hpc stacks, right? (e.g. libsonata is common and then it will be rebuilt twice anyway). Also, I see python version preference in `applications_nse.yaml`. It shouldn't be here but at top level somewhere else? Same about compiler, lapack etc.

Reply:
@pramodk it's still somewhat experimental. But initial thoughts are welcome :)

Also depends on #1239, which is laying some of the groundwork.

I'll see about moving those preferences to a higher level. I merged the 0.16.x release branch recently, which fixed some of the configuration handling of Spack which was blocking parts of this.

Reply:
@pramodk updated the preferences to exclude centrally provided settings…

Reply:
thanks @matz-e ! I will take a look at this shortly!